pymc3/distributions/multivariate.py

```diff
@@ -1329,86 +1368,19 @@ class LKJCorr(Continuous):
-            self.n = n
-            self.eta = eta
-        else:
-            raise ValueError(
-                "Invalid parameter: please use eta as the shape parameter and "
-                "n as the dimension parameter."
-            )
-
-        shape = n * (n - 1) // 2
-        self.mean = floatX(np.zeros(shape))
-
-        if transform == "interval":
-            transform = transforms.interval(-1, 1)
-
-        super().__init__(shape=shape, transform=transform, *args, **kwargs)
-        warnings.warn(
-            "Parameters in LKJCorr have been rename: shape parameter n -> eta "
-            "dimension parameter p -> n. Please double check your initialization.",
-            DeprecationWarning,
-        )
-        self.tri_index = np.zeros([n, n], dtype="int32")
-        self.tri_index[np.triu_indices(n, k=1)] = np.arange(shape)
-        self.tri_index[np.triu_indices(n, k=1)[::-1]] = np.arange(shape)
-
-    def _random(self, n, eta, size=None):
-        size = size if isinstance(size, tuple) else (size,)
-        # original implementation in R see:
-        # https://github.com/rmcelreath/rethinking/blob/master/R/distributions.r
-        beta = eta - 1.0 + n / 2.0
-        r12 = 2.0 * stats.beta.rvs(a=beta, b=beta, size=size) - 1.0
-        P = np.eye(n)[:, :, np.newaxis] * np.ones(size)
-        P[0, 1] = r12
-        P[1, 1] = np.sqrt(1.0 - r12 ** 2)
-        for mp1 in range(2, n):
-            beta -= 0.5
-            y = stats.beta.rvs(a=mp1 / 2.0, b=beta, size=size)
-            z = stats.norm.rvs(loc=0, scale=1, size=(mp1,) + size)
-            z = z / np.sqrt(np.einsum("ij,ij->j", z, z))
-            P[0:mp1, mp1] = np.sqrt(y) * z
-            P[mp1, mp1] = np.sqrt(1.0 - y)
-        C = np.einsum("ji...,jk...->...ik", P, P)
-        triu_idx = np.triu_indices(n, k=1)
-        return C[..., triu_idx[0], triu_idx[1]]
+    bound_args_indices = [-1, 1]
+    rv_op = lkjcorr
 
-    def random(self, point=None, size=None):
-        """
-        Draw random values from LKJ distribution.
+    @classmethod
+    def dist(cls, n, eta, *args, **kwargs):
 
-        Parameters
-        ----------
-        point: dict, optional
-            Dict of variable values on which random values are to be
-            conditioned (uses default point if not specified).
-        size: int, optional
-            Desired size of random sample (returns one sample if not
-            specified).
+        # mean = floatX(np.zeros(shape))
+        n = at.as_tensor_variable(intX(n))
+        eta = at.as_tensor_variable(floatX(eta))
 
-        Returns
-        -------
-        array
-        """
-        # n, eta = draw_values([self.n, self.eta], point=point, size=size)
-        # size = 1 if size is None else size
-        # samples = generate_samples(self._random, n, eta, broadcast_shape=(size,))
-        # return samples
+        return super().dist([n, eta], *args, **kwargs)
 
-    def logp(self, x):
+    def logp(x, n, eta):
```

Comment:
There's a shape mismatch for `logp` in `LKJCov`. Is this because since the parameters `n` and `eta` are 0-dimensional it expects the `logp` to be be of shape equal to `x` which is not the case over here. (It gives out a scalar value, which it should.). Maybe something like https://github.com/pymc-devs/pymc3/pull/4777#discussion_r654755729 is needed here, but for `logp`